### ISSUE TITLE
$repo-path/archive very large and does not shrink

### ISSUE THREAD

**Issue by 2019-05-10** [pgbackrest/pgbackrest#1297]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version: 2.31

2. PostgreSQL version: 12

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each: Debian/SID

4. Did you install pgBackRest from source or from a package? package Debian/SID

5. Please attach the following as applicable:
    - `pgbackrest.conf` file(s)
    - `postgresql.conf` settings applicable to pgBackRest (`archive_command`, `archive_mode`, `listen_addresses`, `max_wal_senders`, `wal_level`, `port`)
    - errors in the postgresql log file before or during the time you experienced the issue
    - log file in `/var/log/pgbackrest` for the commands run (e.g. `/var/log/pgbackrest/mystanza_backup.log`)

7. Describe the issue:

in the folder configured as
repo1-path
pgbackrest creates two subfolders
archive/
backup/

The folder archive/ is basically as big again as the Postgres cluster, and apparently never gets cleaned up.

I am pretty sure, this can be configured somewhere, but after a lot of searching I feeel like running in circles -- the documentation is rather confusing (the Installation site contains the term "archive" more than 200 times, Configuration has "retention" and "retention-archive" without, at least to me, a clear defintion of what is what and how they differ) and seems to presume, I already know the answer to actually find the answer.

I think the docs need a description of the folder structure and their purpose (why is on top of those two there's also /var/spool/pgbackrest/archive/ ?), and how to control the space consumed.

Additionally, there probably should sensible defaults to avoid having the cluster data taking up basically twice its space, _before_ backup already.

pgbackrest.conf:
```
[global]
repo1-path=/home/pgbackrest
repo1-retention-full=4
process-max=4
#archive-async=y
#archive-push-queue-max=400MB
log-level-console=info

#[global:archive-push]
#compress-level=9

[local]
pg1-path=/var/lib/postgresql/12/main
archive-timeout=120
```

**Comment by pgstef** [comment]
Hi,

Indeed, retention options might be confusing in the docs. Sometimes, a few [examples](https://pgstef.github.io/2020/05/25/pgbackrest_preview_tour_of_retention_policy_options.html) might help.

Regarding the directories, `archive` stores the WAL archives (sent with the `archive-push` by PG `archive_command`) and `backup` stores the PG data/tablespaces copied. That design is due to the PITR (continuous archiving) principles.
The `spool` path is only used by for asynchronous operations, as stated in the [docs](https://pgbackrest.org/configuration.html#section-general/option-spool-path).

The amount of WAL files generated depends on your PG activity and configuration. By default, if you take a backup without specifying any retention option, you should get a warning message. It would be a bit complicated to know the "expected" size of WAL archives you could have.

The basic idea here is to keep the WAL files you need to restore your backups or, based on PITR mechanics, to any point in between. The archives will then be expired with their backup. In your configuration, the "old" backups and archives will be automatically expired after your **4th** successful FULL backup. You could use specifically `retention-archive*` option to aggressively expire WAL segments. However, doing so negates the ability to perform PITR from the backups with expired WAL and is therefore not recommended.

So maybe the 1st thing to check here (with your info command for example) would be your actual backups (how many full, diff, incr,..)...

Kind regards

**Comment by 2019-05-10** [comment]
Well, I can't say the examples help much.

Let me rephrase that:
My requirement is to be able to restore to any point in the last 7 days, and use as little as possible disk space 
Currently I intend to do
- a full backup every sunday night
- a diff or incr backup every other night

How should the pgbackrest.conf for that scenario look like and why?

**Comment by pgstef** [comment]
That relates to how you trigger your backups. The usual way of doing it is using cron jobs.

Then, the retention settings in your configuration file will depend on how many full backups you'd like to keep. I would advise at least keeping 2 and then set `repo1-retention-full=2` would be enough. But that number is up to you.

You won't have to configure anything else. Once a 3rd successful `full` backup will happen, the oldest one will expire + all the WAL archives and `diff`/`incr` that depends on that `full` backup will automatically expire too.

**Comment by 2019-05-10** [comment]
Well, those WAL archives are part of my problem -- watching my production DB (for which the backup solution is intended), thos WAL archives currently amount to about 7GB per week -- with ``repo1-retention-full=2`` that would mean to constantly keep between 7 and 15GB WAL archives besides the actual backup files.
Is that the minimum space requirement I have just to live with or is there a configuration to get those 15GB WAL archives down?
After all, shouldn't a daily ``diff/incr`` backup eliminate the need for the WAL archives?

**Comment by dwsteele** [comment]
The WAL is what allows you to play forward from a backup to any point. If you want to use less space you can make fewer backups, which means longer recovery times. You can also use `--repo-retention-archive` to reduce the amount of WAL retained but then you lose your PITR capability for the backups that don't have WAL retained.

My preferences is a weekly full backup (at least 2) and then a rolling 3 differentials. That provides quick PITR for the last three days but still allows PITR for any time in the last three weeks.
